src/main/java/com/redislabs/redisai/Tensor.java

```diff
@@ -0,0 +1,35 @@
+package com.redislabs.redisai;
+
+public class Tensor {
+    private DataType dataType;
+    private long[] shape;
+    private Object values;
+
+    public Tensor(DataType dataType, long[] shape, Object values ) {
+        this.shape = shape;
+        this.values = values;
+        this.dataType = dataType;
+    }
+
+    public Object getValues() {
+        return values;
+    }
+
+    public void setValues(Object values) {
+        this.values = values;
+    }
+
+    public long[] getShape() {
+        return shape;
+    }
+
+    public DataType getDataType() {
+        return dataType;
+    }
+
+    public void setDataType(DataType dataType) {
```

Comment:
@filipecosta90 where are you using the setters? shouldn't we set all fields as final?

Reply:
@gkorland was planning to enable setting tensors via the tensorclass as well.